### DISCUSSION TITLE
New aliens: the Jibauni and Diba

### DISCUSSION THREAD

**Discussion by TheGiraffe3** [endless-sky/endless-sky#10402] [Ideas]
# Overview

This discussion details my plans for a new T1 species, the Jibauni, whose specialty is fuel-based outfits and ships, as well as information about a rival, the robotic Diba. These are controlled by an Alpha, and their specialty is guns that can rotate slightly.

Here is a picture of all my planned systems, besides Jibtrav.

![allspace](https://github.com/user-attachments/assets/4ee5fdc0-2359-4a8d-a8db-d93b02e6474f)

_Note that the Jibauni don't travel to the uninhabited systems unless the player brings them._

Jibtrav, and the surrounding area.

![areajibtrav](https://github.com/TheGiraffe3/Jibauni/assets/145969603/eaeeabcc-f821-4ccc-8392-8042b2e95794)

The galaxy with these systems:

![galaxy](https://github.com/user-attachments/assets/6865c504-fae5-4d9c-bd16-7d4b417bc404)

The plugin with everything in it is [here](https://github.com/TheGiraffe3/Jibauni/tree/Jibauni).
The PR/fork with the Jibauni implemented is [here](https://github.com/Alrodel/endless-sky/pull/2).

**Update 12/17/24: ship sprites are almost ready. I hope to have the Jibauni PRed before Christmas.**
**Update 12/18/24: PR is up! #10867**

## Jibauni

The Jibauni are a species that look like dinosaurs. Their speech, called Jibunese, should be filled with the letter R.
They will probably be using Heliarch translators to communicate.

<details>
<summary>Personality</summary>
The Jibauni will take offense at any little slight; you always have an option to be polite in every mission. If you are not...

Their favorite foods are meat and cheese.
</details>

<details>
<summary>History</summary>

Jibauni written history started sometime in the late 1500s, when the Jibauni became the dominant factor on their home planet of Baitoon. They figured out how to make non-hyperspace capable spacecraft around 2100. They first made their version of the hyperdrive in 2250. After a 50-year period of colonization and good times, one clan in Almanhaq suddenly broke away from the main government, speedily followed by four more. These called themselves the _real_ Jibauni, and scornfully labeled the normal Jibauni the "Lazies." Fighting between the two has been going on ever since then.
Around 2400, a red-pink alien ship took off from a strange green anomaly that had been there ever since the Jibauni could remember, and hailed their homeworld of Baitoon in Jibunese, requesting permission to land. The Jibauni, after a long debate, let the ship land, but posted guards around it, for both their protection and to protect the foreigner from any dissatisfied Jibauni. The next day, the ship was gone, and one of their holy leaders, an Oracle, was missing. Reports came in of the ship passing through Sabibal, but no other traces were found.

In addition, small pirate ships (the biggest possible one being an Argosy) have come through the wormhole every now and then. (There is a set of Jibauni news about this.)
</details>

<details>
<summary>Ships</summary>

The Jibauni ships have built-in slowing resistance, which costs fuel to use and automatically happens. Carried ships have slowing resistance without any cost, to keep them from being unable to move too frequently. I do not think this will cause balance problems with Detainers, because the ships are fighters.
Jibauni ships are faster than a lot of other T1 ships. However, they make up for this by being extremely fragile.
Some of them also have the "fuel mount" attribute, and will be able to carry Fuel-Leaks or Fuelers. See below.

</details>

<details>
<summary>Outfits</summary>

The Jibauni outfits, and indeed the Jibauni in general, are very fuel-based. Their weapons use fuel, and so do their engines. Their generators create fuel. To stop this from being unbalanced, I have made it so that you cannot put them on ships that are not made by the Jibauni.
They also have two special types of weapons, the Fuel-Leaks and Fuelers, which take away and grant fuel, respectively. All of these (except the Leak and er 10) have the attribute "fuel mount" in the negative, and so they cannot be installed on non-Jibauni ships either.
The Fuel ers/Leaks will will be important later in the storyline, but they will not be in an initial PR, especially because sprites for them will be difficult to plan.
</details>

<details>
<summary>"Normal" Storyline</summary>

_Note: in an initial PR the player probably will only be able to do the first few missions, all the way back to Baitoon. The rest will come later._

The storyline starts when landing on Baitoon. A Jibauni named Kanlanmo comes up to the player, and asks them to take him to Taltahard, one jump south-west by west. (Communication is accomplished by the aid of a translation device looted from the Heliarch, though they are not revealed to the player.)
After doing so, the player will take Kanlanmo to the rest of the Normal planets, culminating with taking him back to Baitoon.

30 days later, the player is called to make the rounds again, picking up Jibauni ships as they go. A battle starts in Baitoon while the player is on Kanlil, and `first` `last` must go and save the Jibauni there, being auto-launched if there are still Rebel ships alive. Upon finishing this mission you start receiving a salary of 1,500 credits, which can be adjusted if need be.
At this point, the player must make a decision to either go attack the Rebels in Kateaza, or stay and build up strength. No matter what the player chooses, though, a headstrong Jibauni captain attacks immediately. The player is tasked with going to rescue her, and after landing in the Station of Power, attacking Almanhaq.
In the Almanhaq battle, the Jibauni are beaten off, and must regroup. The player is tasked with bringing a Fury, a Sparrow, and a Scrapper to Jibauni space for study. Fleets are increased on both sides. 
The rest will come later.
</details>

<details>
<summary>"Rebel" Storyline</summary>

In the Jibauni first contact, the player will eventually have the option to decide that the Rebel Jibauni are right and the Normal are wrong. If this happens, the Rebel storyline can begin.
The first contact has a condition for if you are a pirate which won't be added until #6298 is merged. I fully expect that to happen before the Jibauni are ready, but if not, that should be disabled; it is meant to test if you have swizzle 6 or not.
To begin with, you are tasked with fetching some alien warships (3 Scrappers), which take part in the Battle for Baitoon.
The player leads the assault, but is driven back by sheer numbers. You must then defend Almanhaq from a Jibauni counterattack, but with the whole Rebel armada on your side, this is quite easy. Landing on Almanhaq, you are given a Rebel license and a salary of 1,200 credits. (This also can be adjusted.) Fleets on both sides are increased.

</details>

<details>
<summary>Side Mission Chains</summary>

There is a culture conversation on Kanlil which can will up turning into a small mission chain where you transport a Jibauni, help the Jibauni get a new ship, and then escort that ship to Solace for a "tour". Hopefully there will be other missions like this - the Jibauni are interested in stealthily mapping human space.
</details>

<details>
<summary>Ties with other factions</summary>

Diba: Hostile because of the kidnapping.

Each other (Rebel/Normal fight): Hostile because of revolution and name-calling.

Pirates: Hostile, because why would they like each other? Pirates are pirates...

The lore reason for the Quarg not living nearby to keep an eye on the Jibauni is that they are still bitter about the thievery of their rings by the Heliarch.

The Drak don't really care right now, leaving it to the Quarg to decide what to do.

I'm not sure of a good reason the Pug wouldn't intervene in any Jibauni affairs, because "the Pug are used enough already" definitely doesn't make sense lore-wise.

</details>

## Diba

The Diba are a robot faction under the control of a very bad "person," a. k. a. an Alpha...

<details>
<summary>History</summary>

The Diba are a faction of robots controlled by an Alpha who escaped from the original scattering in the Alpha Wars. It went through the wormhole in Jibtrav, landing on Huyya, where it kidnapped a Jibauni Oracle and went to an uninhabited system, whose stars move slightly. It has been there ever since, creating robotic warships, and getting ready to fight when the Jibauni come to get their revenge.
The Oracle is also still with it, trying to find a good way to escape. This will be a pivotal point later on in the story.
</details>

<details>
<summary>Ships</summary>

One of the reasons their ships are interesting follows. They have "turrets," but the turrets have limited rotation arcs and so are basically slightly movable guns, which is what they are in lore.
The ships have built-in engines, energy generation, and the attribute "diba gun port" with a negative number set to however many "turrets" they have, being 1, 2, 4, 8, and 16 respectively for the Llama, Ithquaya, Arbquaya, Tamquaya, and Tahquaya.
All Diba ships also have one bunk, so the Alpha can escape any time.
The Llama is a fighter whose single turret has a limited rotation arc of 10 degrees and which has 100 cargo space. I can nerf this if desired, but the other Diba ships have a maximum cargo of 8, so I'm not sure it's too bad.
Thanks to Petersupes for the sprite.
</details>

<details>
<summary>Outfits</summary>

The Diba have two outfits: the Diba Laser Turret, which is an improved version of the human Heavy Laser Turret, and the Diba Anti-Missile, which is an improved version of the Heavy Anti-Missile Turret. They both have the attribute "diba gun port" 1, which means that all Diba outfits will stay on the ship they came on. They are also unlootable, so they can never be on a non-Diba ship.
</details>

<details>
<summary>Storyline</summary>

There will be none, unless an Alpha storyline is added someday. Because you're a normal human, the Alphas hate you too much to let you do anything related to the Diba "empire".
However, a way to get further in the Jibauni storyline may be to destroy the Diba leader's personal ship. 
</details>

<details>
<summary>Ties with other factions</summary>

I'm not sure that "ties" is the right way to put it...

The Diba are hostile with every faction except the Alphas.
</details>

## Balance

Basically, the Jibauni are agile and fragile. T0.9 HP wise but quite powerful in matters of fuel, which is balanced by the fact that none of their good outfits can be installed on non-Jibauni ships.

The Diba are around T1.1 militarily, and have slightly shiftable gun ports. They do not specialize.

## Closing thoughts

~~Like the Naltok, the Jibauni are not yet ready for PR to vanilla. They have no outfit sprites that have blend files, and I'm not good at Blender.~~

I've started making Jibauni outfit sprites. We'll see how good they are...
@hrichharms is making ship sprites for the Jibauni.

**Comment by hrichharms** [comment]
I would be interested in helping with some Jibauni images. How would be the best way to contribute?

**Reply by TheGiraffe3** [reply]
Just make a PR to the repository with your suggested sprites. Thanks for offering!

**Reply by hrichharms** [reply]
I am curious as to what sort of design language you had in mind for them if you had any existing thoughts. Are you in the discord?

**Reply by TheGiraffe3** [reply]
I haven't really had any plans for how they should look, though as I mentioned, they should look fast.
I'm not in the Discord, at least not yet.

**Reply by reticent-rem** [reply]
I really like thematic designs -- since the Jibauni look like dinosaurs perhaps their ships could be inspired by dinosaur skulls? Below are the skull profiles of a Velociraptor, Tarbosaurus bataar, and T-Rex; which I think actually look like streamlined ships from above.


![images](https://github.com/user-attachments/assets/28a0209e-6a30-4eb8-ba14-819cd10c8135)
![6-Figure2-1](https://github.com/user-attachments/assets/81da2e88-9a8d-49f8-9b7e-f9c764428fd5)

**Reply by TheGiraffe3** [reply]
That could be a good design philosophy. A2 might look somewhat similar to the Arrow though... but yes, I think that's a great idea!
Maybe though without the colors.

**Reply by hrichharms** [reply]
I like the organic yet streamlined look! Going with the narrative consequences, I would be curious as to what sort of interesting cultural factors amongst them led them to design their ships to resemble their own skulls. Maybe the alien invasion was particularly devastating and left their society with a different perspective on death than human society?

**Reply by TheGiraffe3** [reply]
They're using whatever shapes "come to mind" (pun intended).
Also, the Jibauni like symmetrical shapes, and skulls were the best thing they could find at the time.

**Reply by hrichharms** [reply]
That could work, though I still think that it may be somewhat puzzling for players to see a species that designs their ships to look like skulls without some kind of deeper cultural reason behind it. That being said I really like the Velociraptor skull. Will try to get a first draft of a sprite cooked up in the next couple days to test out the design philosophy.

**Reply by reticent-rem** [reply]
Spitballing again, but since "the Jibauni will take offense at any little slight" and have been fighting so many wars, maybe they have a bit of a warrior culture and collected the skulls of their enemies in the past. Originally this meant the skulls of other Jibauni, and these skulls became status symbols and signs of combat prowess. This cultural aspect continued as they reached the stars, with ships designed after their skulls to demonstrate status and prowess. That could even explain some variations in ship designs, with the more expensive and higher end ships more directly looking like skulls, and cheaper/worse ships looking like more blocky (or just not as detailed) skull imitations.

**Reply by TheGiraffe3** [reply]
That would be an interesting lore reason. I could add a question about that to the bunch of questions at the beginning of the second mission.

However, I'm unsure about lower-end ships (like the Yankul) looking much worse, because that could be considered another slight to those who don't have as much money...

Though when the wars actually started, they weren't fighting _each other_, but rather dragons.
So maybe the two main warships (the Nathar and the Yiyharsi) could be shaped more like dragon skulls, the interceptors (Furus and Saghir) be a mix, and the other ones like dinosaurs?

**Reply by hrichharms** [reply]
I like the dragon skulls idea. Could be result in some interesting interactions with the Sheragi during the story.

Ships shaped like the skulls of other species of dinosaurs may also have a link to a hunting culture in their history. Maybe the hunting of other dinosaurs was considered an honourable substitute for enemy skulls during times of peace?

**Reply by TheGiraffe3** [reply]
Then we'll have to decide what happened to those other dinosaurs later on in Jibauni history.

Probably they were moved to Owiay, because the Jibauni stopped hunting them and thought about species conservation instead. The other dinosaurs were vegetarians, so they can survive perfectly on Owiay, and meanwhile the Jibauni have figured out how to create synthetic meat.

**Reply by TheGiraffe3** [reply]
I added a mention of that in https://github.com/TheGiraffe3/Jibauni/commit/c5f7b2ec140ca4fbdfe74b3b90057af46fd13851 (with a few other changes). Does that seem like the situation is sufficiently described, on line 310?

**Reply by reticent-rem** [reply]
I think that addresses it, and have some thoughts on lines 314/315.
1. Kanlamo mentions their ships are shaped like dinosaur skulls as a "good memorial to all our fallen friends", but also like dragon skulls because "they were good at fighting." To me this reads like they make dino skull ships primarily out of love/memory of close friends, but dragon skull ships out of respect for a capable enemy. Why did the reasoning change from a sense of cherished remembrance to an enemies respect? Or it was still about foes throughout time then maybe something like "good memorial to all our fallen friends" to "good memorial to all our respected foes"?
2. "Other non-combat ships are in the shape of other dinosaurs' which we used to eat skulls." makes it sound like they eat the skulls of other dinosaurs. Maybe "Other non-combat ships are inspired by the skulls of dinosaurs we used to hunt.'"

**Reply by TheGiraffe3** [reply]
I've updated it with those suggested changes. Thanks!

**Comment by Arachi-Lover** [comment]
A couple of points on things that I don't think work all that well:

`Since the 2800s, the Jibauni have also found Heliarch ships coming in and mining asteroids from Quinlo. Because the Heliarch weapons/ships (even small ones) are so powerful, the Jibauni have lost many people while trying to disable their opponents. However, on rare occasions they manage to disable the Heliarch, and loot things from their ships. This is mainly translation devices, because they can't figure out how to get the Jump Drive (or anything else) out. Also, every time a Heliarch ship is disabled, it sends a distress call, and more Heliarch come in to help, repairing the disabled ship, which still has its Jump Drive.`

So the Heliarchs have grown tremendously paranoid over their remaining JDs, as they have had over the millennia many failed expeditions out of Coalition space where they've lost probably in total some couple dozen JDs. They would still try and keep tabs on other areas, as they do with checking in periodically with Poisonwood, in the case of human space, but these are brief, single-ship (Punisher, as it's the one that's least likely to be beaten before it can flee) visits, in peaceful manner, as they really don't want any more reason to believe they'll be attacked out of Coalition space and have yet another of their JDs at risk.
There's also the matter that the Heliarchs using JD-equipped ships to mine things elsewhere would be a very odd choice, as Coalition space itself has numerous resource-rich systems, and even if the mineables in question were say Yottrite, which they value a lot, they still don't risk going out on their own for that even though there's Saquergen, a nova system with Yottrite asteroids, really close to their Arach-Human border. I don't think they'd be going through all this trouble to mine asteroids.
As you mention the distress call thing, I'd just like to point out that the Heliarchs rely on the hyperspace network for communication, so they would have no means of calling for help from a system that isn't linked to Coalition space.
Overall, I think the bit for the Heliarch translators can be simply something the Jibauni demanded as part of some form of agreement (ie something akin to Poisonwood, where the Heliarchs periodically come in to gather information), maybe they could also work out receiving some help with less restricted Coalition tech pieces.

`The lore reason for the Quarg not living nearby to keep an eye on the Jibauni is that they are still bitter about the thievery of their rings by the Heliarch.
The Drak don't really care right now, leaving it to the Quarg to decide what to do.`

I accept the reason for the Quarg not being there yet, as it'd probably tip off the Heliarchs more than they already are. I'm not sure on the Drak not really caring and letting the Quarg decide though, mainly as it seems to contradict the previous bit? If the Quarg aren't around, how do the Drak expect the Quarg to keep an eye there and decide things?

`I'm not sure of a good reason the Pug wouldn't intervene in any Jibauni affairs, because "the Pug are used enough already" definitely doesn't make sense lore-wise.`

I think we need to be careful with continuing to refuse the Pug to be involved anywhere, as that's already being remedied by many of the recent species additions and planned stories, so we could end up with the opposite case, where the Pug are scarcely involved at all throughout the game's areas, which would also be a problem. I'm not saying you absolutely need to involve the Pug, but I don't think we need to discard the possibility immediately as we were doing at one point.

`The Diba are a faction of robots controlled by an Alpha who escaped from the original scattering in the Alpha Wars. It went through the wormhole in Jibtrav, landing on Huyya, where it kidnapped a Jibauni Oracle and went to an uninhabited system, whose stars move slightly. It has been there ever since, creating robotic warships, and getting ready to fight when the Jibauni come to get their revenge.
The Oracle is also still with it, trying to find a good way to escape. This will be a pivotal point later on in the story.`

It seems kind of crazy that this one Alpha would be able of establishing all the industry needed to make many many robotic warships to the point they can fight off another faction. Should it not be a larger Alpha group, with say a fleet instead of the one ship?
Also, is this Oracle kidnapping thing with the Alphas related to the Oracle AI the Alphas presented the Syndicate with, or is it just a coincidence in the names?

Also, are the Diba like, a species the Alpha(s?) took control of, or were they wholly made by the Alpha? I guess what I'm asking is, did the Diba exist before the Alphas? Also, are they supposed to be like, a bunch of droids walking around planets, and also the warships, or just the robot warships kind of like the Automata?

`All the Sheragi stuff`

Honestly I don't think it is a good idea to un-extinction the Sheragi. They out of all the extinct species are the ones that get the most focus, and I think it'd muddy all of the weight behind that, and especially the Emerald Sword mission chain, if it turned out there was a pocket group of Sheragi hiding out somewhere. If these guys existed, would the Drak/Archons somehow not know? Why would the Archon watching over the Emerald Sword speak/give visions of the Sheragi's fall in such a manner if the Sheragi were still around? I'm also heavily opposed to the idea proposed in MZ's lore doc, of the Pug having saved a group of Sheragi, for these reasons.

Looking at this as a whole, I don't think the Sheragi are much integral to your story concepts, so my 2 cents would be to avoid the Sheragi altogether, and instead focus on the Jibauni dealing with the various groups that they have to interact with (player, Pirates, Heliarchs) and the conflict with the Diba/Alphas.

**Reply by TheGiraffe3** [reply]
> about the Heliarch

Sounds good.

> Drak/Quarg

The Drak just delegated it and then focused elsewhere.
I'm thinking of the fact that there is an implied Archon in Ildaria, and other than that there are none south of Sol.
I suppose I could add another one near the Jibauni, but in my opinion there shouldn't be that many, hence why I've avoided adding one, at least yet.

> Pug

What I had seen was that the Pug were overused at the moment. However, I wouldn't be opposed to including them with Jibauni content, at all.

Though wouldn't this more heavily imply that the Pug gave the Arachi the hull repair which started the war with the Quarg? Then again, the Jibauni didn't start until 4500 years after that, so it could work.

> Diba

He has had two hundred years to do so in. However, there could be more.
The Diba are more like the Sestor and Mereti, although in this case they have an active creator who is directly planning their strategy.
The Oracle is a coincidence - in fact, when thinking of it I had forgotten about the FW Oracle.
I could change it to "Prophet" if people want.

> Sheragi

I was planning on making the Sheragi a larger plot point. And I've done quite a bit of work on them, so before removing them altogether I would like to get a second opinion - though you're right, it would just increase the strangeness of having no Archon/Quarg intervention.

**Reply by Saugia** [reply]
In the past we've generally made the point to have the Sheragi remain an extinct race. The archaeology mission they're involved in is a stretch of twist and turns to reveal them as a bygone specialty, rewarding the player of their time with an interesting story from the past and the trophy relic that comes with it. Thematically it's better to keep them ancient, rather than try and pull every single race in Endless Sky up to be plot relevant and exist in some fashion.

MZ had originally considered there to be a Sheragi sub-faction somehow held by the Pug in the past as well, but that was widely un-canonized at large for the same above reasons by creators and developers alike. To have something say, "But in reality, they were never gone, and are a big deal here", would take away from the story in the archaeology missions and the value of them.

**Reply by TheGiraffe3** [reply]
They've been taken out.

**Comment by TheGiraffe3** [comment]
Would like to note that other than outfit sprites, the Jibauni are now entirely ready to PR. If anybody would make them for the Jibauni, that'd be much appreciated.

**Reply by SpaceCaptainBill** [reply]
What outfits need sprites?

**Reply by TheGiraffe3** [reply]
Oops, I forgot to mark that comment as outdated.
I've created sprites for all the outfits at this point.
